### PR TITLE
TST: tweak Hypothesis configuration and idioms

### PR DESCRIPTION
**Pull request by Zac-HD**
#19831 saw a flaky test, where a slow VM on Travis [went over the default deadline](https://travis-ci.org/pandas-dev/pandas/jobs/448750638#L2895).  This PR therefore configures the per-test-case deadline to a higher value, and disables the global timeout (which is deprecated - we know it's confusing but haven't bumped a major version in years...).

I've also moved a custom strategy from a method on the class to a standalone function, which should avoid some confusion as to where the `draw` argument comes from!

### CODE REVIEW

File: pandas/tests/frame/test_apply.py [hunk 1]
```diff
@@ -1142,20 +1156,7 @@ def test_agg_cython_table_raises(self, df, func, expected, axis):
...
-                max_value=Timestamp.max.floor("D").to_pydatetime().date(),
-            ).map(Timestamp)
-        )
-        periods = draw(integers(0, max_length))
-        freq = draw(sampled_from(list("BDHTS")))
-        dr = date_range(date, periods=periods, freq=freq)
-        return pd.DatetimeIndex(list(dr))
-
-    @given(index=indices(5), num_columns=integers(0, 5))
+    @given(index=indices(max_length=5), num_columns=integers(0, 5))
```
Comment (gfyoung) [inline pandas/tests/frame/test_apply.py]: Generally, we would prefer that separate changes go into separate PR's, but these are small enough that it should be okay.

Comment (Zac-HD) [reply pandas/tests/frame/test_apply.py]: It's *kinda* related in that this is input to the test that was timing out, but point taken - I'll try to split future changes like this into multiple PRs :smile:

**Comment by Zac-HD** [comment]
CC @TomAugspurger @jreback FYI.

**Comment by TomAugspurger** [comment]
Thanks!
